qiskit/dagcircuit/dagcircuit.py

```diff
@@ -466,7 +467,8 @@ def _check_wiremap_validity(self, wire_map, keymap, valmap):
                 raise DAGCircuitError("invalid wire mapping key %s" % kname)
             if v not in valmap:
                 raise DAGCircuitError("invalid wire mapping value %s" % vname)
-            if type(k) is not type(v):
+            # support mapping from AncillaQubit to Qubit
+            if not (isinstance(k, type(v)) or isinstance(v, type(k))):
```

Comment:
This fixes the issue, but it looks like this is necessary because when calling `to_instruction`, `AncillaQubits` are replace with regular `Qubits` in the `definition`. Is that the right behavior?